src/libraries/System.Reflection.Metadata/src/System/Reflection/Metadata/TypeNameParserHelpers.cs

```diff
@@ -438,12 +438,11 @@ internal static bool TryDive(TypeNameParseOptions options, ref int depth)
             return !IsMaxDepthExceeded(options, depth);
         }
 
-#if SYSTEM_REFLECTION_METADATA
         [DoesNotReturn]
         internal static void ThrowInvalidOperation_NotSimpleName(string fullName)
         {
+            // This message is defined in both System.Private.CoreLib and System.Reflection.Metadata
```

Comment:
The pattern used elsewhere is to assert or throw an exception without message under `SYSTEM_PRIVATE_CORELIB` ifdef. E.g. check `SYSTEM_PRIVATE_CORELIB` ifdef earlier in this file.

Reply:
I'm not entirely sure I get the utility of the "simple name" check. I would imagine users can place `typeof(int[])` as the type, which seems like it would match the notion of "array" mentioned in the check for `IsSimple` in the `TypeName` class.

Reply:
The method will replace the assembly name for the top level TypeName only. It does not produce the expected result for non-simple names, and it is not clear what the expected result should be once you get to generics.